addons/portal/models/mail_message.py

```diff
@@ -72,11 +73,16 @@ def _portal_message_format(self, properties_names, options=None):
         if 'attachment_ids' in properties_names:
             properties_names.remove('attachment_ids')
             attachments_sudo = self.sudo().attachment_ids
-            attachments_sudo.generate_access_token()
             related_attachments = {
-                att_read_values['id']: att_read_values
-                for att_read_values in attachments_sudo.read(
-                    ["access_token", "checksum", "id", "mimetype", "name", "res_id", "res_model"]
+                att_read_values["id"]: {
+                    **att_read_values,
+                    "limited_access_token": limited_field_access_token(att, "raw"),
+                }
+                for att, att_read_values in zip(
+                    attachments_sudo,
```

Comment:
Is it normal that before we used the id and now the record?

Reply:
The data structure is the same as before. The only change is that the `limited_access_token` value is added to each attachment data instead of the `access_token` field. 
```py
related_attachments = {
    1: {
        'checksum': 'xyz456',
        'id': 1,
        'limited_access_token': 'some_value'
        'mimetype': 'image/jpeg',
        'name': 'example.jpg',
        'res_id': 123,
        'res_model': 'mail.message'
    },
    2: {
        'checksum': 'ghi789',
        'id': 2,
        'limited_access_token': 'some_value'
        'mimetype': 'application/pdf',
        'name': 'document.pdf',
        'res_id': 456,
        'res_model': 'mail.message'
    }
}
```

Reply:
My bad ^^